src/pages/App/index.js

```diff
@@ -42,7 +43,9 @@ class App extends React.Component {
     firebase.auth().onAuthStateChanged(async user => {
       if (user) {
         const tokenResult = await user.getIdTokenResult();
-        const { claims } = tokenResult;
+        const { claims, token } = tokenResult;
+
+        ApiConfigStore.setToken(token);
```

Comment:
might also have to call this in setClaims

Reply:
we could have a state field that's token instead of userRoles - what do you think

Reply:
setClaims doesn't have access to tokenResult.token though. Also, I use userRoles for the RBAC HOCs so I'd rather it still be there. We can always just have a userToken state field in addition to the other two.

Reply:
yeah i see ur point about the userRoles, but after a user logins you should be able to get their token

Reply:
so then we can add the userToken as a state field

Reply:
Yes that's fine. What about the calling setToken() in setClaims that you said in your first comment of this thread?

Reply:
yeah cuz after someone signs in we should set the token in setClaims instead of waiting for the firebase authListener to do it

Reply:
i ran into some weird issues before doing this

Reply:
Should I put userToken into userClaims then, since it's an object? Also, if I were to call setToken in setClaims, do I just get the token from this.state.userClaims?